src/libraries/System.IO.Ports/src/System.IO.Ports.csproj

```diff
@@ -106,17 +108,10 @@
     <Compile Include="$(CommonPath)Interop\Windows\Kernel32\Interop.FileOperations.cs"
              Link="Common\Interop\Windows\Kernel32\Interop.FileOperations.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsLinux)' == 'true'">
-    <Compile Include="System\IO\Ports\SerialPort.Linux.cs" />
-  </ItemGroup>
-  <ItemGroup Condition=" '$(TargetsOSX)' == 'true' ">
-    <Compile Include="System\IO\Ports\SerialPort.OSX.cs" />
-  </ItemGroup>
-  <ItemGroup Condition=" '$(TargetsFreeBSD)' == 'true' ">
-    <Compile Include="System\IO\Ports\SerialPort.FreeBSD.cs" />
-  </ItemGroup>
-  <ItemGroup Condition="'$(TargetsLinux)' == 'true' or '$(TargetsOSX)' == 'true' or '$(TargetsFreeBSD)' == 'true'">
```

Comment:
@wfurt can you review as well? Do you still remember the differences between all of those OSes?

Reply:
The only observable difference that I noticed is that the GetPorts implementation is different. Everything else is just the same.

Reply:
yes, that is correct. The GetPortNames() detects available (standard) serial ports and the device names and methods are different. The code does not use any platform specific API so moving that to single file as separate functions is ok. The old style may make it more visible but I don't think it maters in practice.